sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorConnection.java

```diff
@@ -48,9 +46,8 @@
     private final ConcurrentMap<String, SessionSubscription> sessionMap = new ConcurrentHashMap<>();
 
     private final AtomicBoolean isDisposed = new AtomicBoolean();
-    private final DirectProcessor<AmqpShutdownSignal> shutdownSignals = DirectProcessor.create();
-    private final FluxSink<AmqpShutdownSignal> shutdownSignalsSink = shutdownSignals.sink();
-    private final ReplayProcessor<AmqpEndpointState> endpointStates;
+    private final Sinks.One<AmqpShutdownSignal> shutdownSignalSink = Sinks.one();
```

Comment:
Based on https://projectreactor.io/docs/core/release/api/reactor/core/publisher/DirectProcessor.html the closest Sinks equivalent would be `Sinks.many().multicast().directBestEffort()`

Reply:
Yeah. There is ever going to be a single signal though. So since we had shipped the connection API, under the hood, it would be a single sink but exposed as that.